pkg/infrastructure/nutanix/clusterapi/clusterapi.go

```diff
@@ -146,44 +149,126 @@ func (p Provider) Ignition(ctx context.Context, in infracapi.IgnitionInput) ([]b
 	}
 	imgReq.Metadata = imgMeta
 
+	var imgUUID string
+	timeout := 10 * time.Minute
+	// wait for successful creation of the bootstrap image object in PC
+	if err = wait.ExponentialBackoffWithContext(ctx, wait.Backoff{
+		Duration: time.Minute * 2,
+		Factor:   float64(1.0),
+		Steps:    5,
+		Cap:      timeout,
+	}, func(ctx context.Context) (bool, error) {
+		uuid, e1 := createImage(ctx, nutanixCl, imgReq, imgName)
+		if e1 != nil {
+			logrus.Errorf("failed to create the bootstrap image object %s in PC: %v", imgName, e1)
+			// no need to retry if the error code is 401 or 403
+			if strings.Contains(e1.Error(), "\"code\": 401") || strings.Contains(e1.Error(), "\"code\": 403") {
+				return false, e1
+			}
+
+			// delete the image object if uuid is not empty
+			if uuid != "" {
+				if e2 := deleteImage(ctx, nutanixCl, uuid); e2 != nil {
+					logrus.Errorf("failed to delete image object %s (uuid: %s): %v", imgName, uuid, e2)
+				}
+			}
+			return false, nil
+		}
+
+		imgUUID = uuid
+		return true, nil
+	}); err != nil {
+		if wait.Interrupted(err) {
+			err = fmt.Errorf("timeout/interrupt to create the bootstrap image object %s in PC within %v: %w", imgName, timeout, err)
+		} else {
+			err = fmt.Errorf("failed to create the bootstrap image object %s in PC: %w", imgName, err)
+		}
+
+		return in.BootstrapIgnData, err
+	}
+	logrus.Infof("Successfully created the bootstrap image object %s (uuid: %s)", imgName, imgUUID)
+
+	// wait for successful uploading the bootstrap image data from file imgPath to the image object in PC
+	timeout = 15 * time.Minute
+	if err = wait.ExponentialBackoffWithContext(ctx, wait.Backoff{
+		Duration: time.Minute * 3,
+		Factor:   float64(1.0),
+		Steps:    5,
+		Cap:      timeout,
+	}, func(ctx context.Context) (bool, error) {
+		e1 := uploadImageData(ctx, nutanixCl, imgName, imgUUID, imgPath)
+		if e1 != nil {
+			logrus.Errorf("failed to upload the bootstrap image %s data: %v", imgName, e1)
+			// no need to retry if the error code is 401 or 403
+			if strings.Contains(e1.Error(), "\"code\": 401") || strings.Contains(e1.Error(), "\"code\": 403") {
```

Comment:
Nit: you can use the `` avoid the escaping:
```suggestion
			if strings.Contains(e1.Error(), `"code": 401`) || strings.Contains(e1.Error(), `"code": 403`) {
```
Here it doesn't much difference but it can be useful for longer strings with lots of escaping.